ablySpec/TestUtilities.swift

```diff
@@ -414,6 +414,13 @@ class TestProxyTransport: ARTWebSocketTransport {
         super.receive(msg)
     }
 
+    func closeAbruptly() {
+        // Inject a ProtocolMessage.CLOSE
```

Comment:
I don't think this is a valid mock, because if the real transport does indeed close abruptly it won't make up a ProtocolMessage.CLOSE. I guess the connection should be notified of the close somehow (with a callback, I guess) and then check the current state, so that if it is CLOSING it transitions to CLOSED.